### ISSUE TITLE
Change default dir context vaidator to have tcp timeout

### ISSUE THREAD

**Issue by ldappidydoo** [spring-projects/spring-ldap#567]
Hello, it would be nice if the default dir context validator set a short tcp timeout in addition to a short ldap timeout. I find using the context validator, when it fails to validate it is due to the higher tcp timeout it encounters from the sun jndi read-timeout -- not the 500ms search control timeout. It never gets as far as actually executing the test search.

**Comment by jzheaux** [comment]
I think timeouts can be configured in the `DirContext` that gets passed in. So, I wonder if this is more a matter of configuring your `ContextSource`.

If not, what are you thinking that support for this via `DefaultDirContextValidator` would look like?

**Comment by ldappidydoo** [comment]
Apologies, been a long time since I visited this. I think I found a workaround that suited me via pool configuration which would discard a connection if it exceeded a certain age before testing it. I was grappling with a firewall which would let connections die silently in some way.

Without the pooling, I think my rationale for this request was to have a shorter timeout on the context validator than the context itself. As what would happen to me is the act of connection validation would take a long time. My goal was to validate and discard connections quickly. The medicine was as bad as the virus.
